core/health/src/main/java/io/thorntail/health/impl/HealthServlet.java

```diff
@@ -1,110 +1,9 @@
-import org.eclipse.microprofile.health.HealthCheckResponse;
-
 /**
  * Created by bob on 1/17/18.
  */
-public class HealthServlet extends HttpServlet {
-
-    private static final Map<String, ?> JSON_CONFIG = new HashMap<String, Object>() {{
-        put(JsonGenerator.PRETTY_PRINTING, true);
-    }};
-
-    @Override
-    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
-        OutputStream out = resp.getOutputStream();
-        JsonWriterFactory factory = Json.createWriterFactory(JSON_CONFIG);
-        JsonWriter writer = factory.createWriter(out);
-
-        JsonObject payload = jsonObject();
-        String outcome = payload.getString("outcome");
-        if ( outcome.equals(HealthCheckResponse.State.DOWN.toString())) {
-            resp.setStatus(503);
-        }
-
-        writer.writeObject(payload);
-        writer.close();
-    }
-
-    private JsonObject jsonObject() {
-        JsonArrayBuilder results = Json.createArrayBuilder();
-        HealthCheckResponse.State outcome = HealthCheckResponse.State.UP;
-
-        for (HealthCheck check : checks) {
-            if (check == null) {
-                continue;
-            }
-            JsonObject each = jsonObject(check);
-            results.add(each);
-            if (outcome == HealthCheckResponse.State.UP) {
-                String state = each.getString("state");
-                if (state.equals("DOWN")) {
-                    outcome = HealthCheckResponse.State.DOWN;
-                }
-            }
-        }
-
-        JsonObjectBuilder builder = Json.createObjectBuilder();
-
-
-        builder.add("outcome", outcome.toString());
-        builder.add("checks", results);
-
-        return builder.build();
-    }
-
-    private JsonObject jsonObject(HealthCheck check) {
-        return jsonObject(check.call());
-    }
-
-    private JsonObject jsonObject(HealthCheckResponse response) {
-        JsonObjectBuilder builder = Json.createObjectBuilder();
-        builder.add("name", response.getName());
-        builder.add("state", response.getState().toString());
-        response.getData().ifPresent(d -> {
-            JsonObjectBuilder data = Json.createObjectBuilder();
-            for (Map.Entry<String, Object> entry : d.entrySet()) {
-                Object value = entry.getValue();
-                if (value instanceof String) {
-                    data.add(entry.getKey(), (String) value);
-                } else if (value instanceof Long) {
-                    data.add(entry.getKey(), (Long) value);
-                } else if (value instanceof Boolean) {
-                    data.add(entry.getKey(), (Boolean) value);
-                }
-            }
-            builder.add("data", data.build());
-        });
-
-        return builder.build();
-    }
-
-    @Inject
-    @Health
-    private Instance<HealthCheck> checks;
+public class HealthServlet extends io.smallrye.health.SmallRyeHealthServlet {
```

Comment:
why not import it?

Reply:
You are right, right now there's no obvious need for it. I thought it would have an option to have some Thorntail specific in Thorntail own servlet if needed. The other thing is that it is not clear yet if SmallRye Health should have its own servlet, it has at the moment but I know there are concerns it won't fit into Undertow HttpHandler...

Reply:
yeah, I had trouble making it work with servlet in metrics, I ended up doing a implementation-agnostic request handler in SmallRye and then creating a servlet in v4 and undertow handler in v2.
Either way, I think it would be good to remove this class now if it's not needed. It will be easy to add it back in future if needed.

Reply:
Sure, have just removed the servlet, thanks